### PR TITLE
Provide swapchain to init

### PR DESCRIPTION
**Pull request by Lunahria**
Allows you to, for example, create textures for use in plugins.

### CODE REVIEW

File: codegen/src/lib.rs [hunk 1]
```diff
@@ -176,7 +181,7 @@ pub fn arcdps_export(item: proc_macro::TokenStream) -> proc_macro::TokenStream {
 
             fn load() -> &'static ArcDpsExport {
                 let mut export = &EXPORT;
-                let res: Result<(), Box<dyn ::std::error::Error>> = #init;
+                let res: Result<(), Box<dyn ::std::error::Error>> = unsafe { #init };
```
Comment (greaka) [inline codegen/src/lib.rs]: Why is the unsafe needed here?

**Review by greaka — COMMENTED** [review]
Thanks for contributing :slightly_smiling_face: 

This is a breaking change, so I will need to bump the minor version of this crate, but I have a few changes that I'd like to see first and a few open question.

File: codegen/src/lib.rs (hunk 1 above)
Comment (Lunahria) [reply codegen/src/lib.rs]: Thank you for looking at it!
It is indeed a breaking change and should only be added to a new version.

> Why is the unsafe needed here?

I was hoping you would catch that so I can get your recommendation on how we can avoid the:
**error[E0133]**: use of mutable static is unsafe and requires unsafe function or block
compiler error that arises when the unsafe block is removed.

I'm not good at all with macros and pretty new to Rust so I'm open to all suggestions!

Comment (greaka) [reply codegen/src/lib.rs]: Thanks, if you're okay with it, I would push some changes, merge and then create a new release :)

Comment (Lunahria) [reply codegen/src/lib.rs]: By all means. Thank you!

**Comment by greaka** [comment]
New version released :slightly_smiling_face:

**Pull request merged**